src/main/java/hng_java_boilerplate/payment/Utils.java

```diff
@@ -0,0 +1,49 @@
+package hng_java_boilerplate.payment;
+
+import hng_java_boilerplate.payment.dtos.responses.PaymentResponse;
+import hng_java_boilerplate.payment.entity.Payment;
+import hng_java_boilerplate.user.entity.User;
+import hng_java_boilerplate.user.exception.UserNotFoundException;
+import hng_java_boilerplate.user.service.UserService;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.stereotype.Component;
+
+import java.util.UUID;
+
+@Component
+public class Utils {
+
+    private final UserService userService;
+
+    public Utils(UserService userService) {
+        this.userService = userService;
+    }
+
+    public String generateTransactionReference() {
+        return "TRANS_" + UUID.randomUUID();
+    }
+
+    public User validateLoggedInUser() {
+        User user = userService.getLoggedInUser();
+        if (user != null) {
+            return user;
+        } else {
+            throw new UserNotFoundException("User not authorized");
```

Comment:
shouldn't this be unauthorized exception 401?

Reply:
The method is already throwing a UserNotFound; The if else is redunctant and if user is null will never run